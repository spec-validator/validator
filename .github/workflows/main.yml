name: CI

on:
  push:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@master

      - name: Configure node
        uses: actions/setup-node@master
        with:
          node-version: "12.x"

      - name: Install
        run: yarn install

      - name: Lint
        run: yarn lint

      - name: Test
        run: yarn test

      - name: Build
        run: yarn build

      - name: Prepare GITHUB_TOKEN
        run: echo "//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_TOKEN }}" > ~/.npmrc

      - name: Prepare GIT
        run: |
          git config --global user.email "<>"
          git config --global user.name ${GITHUB_ACTOR}
      - name: Publish
        run: yarn publish

      - name: Publish Storybook
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          mv ./storybook-static /tmp
          cd /tmp/storybook-static
          git init
          git checkout -b gh-pages
          echo spikard.gurunars.com > CNAME
          git add .
          git commit -am "Storybook build ${GITHUB_SHA}"
          git push --force https://${GITHUB_ACTOR}:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git gh-pages
