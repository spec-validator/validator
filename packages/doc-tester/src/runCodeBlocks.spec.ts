import main from './runCodeBlocks'

jest.mock('fs', () => ({
  readFileSync: () => [
    '# Title',
    '```ts',
    'const meaning = 42',
    '```',
  ].join('\n'),
  __esModule: true,
}))

jest.mock('./discover', () => ({
  default: () => ['ONE.md', 'TWO.md'],
  __esModule: true,
}))

const execSync = jest.fn()

jest.mock('child_process', () => ({
  execSync: (cmd: string, props: unknown) => execSync(cmd, props),
  __esModule: true,
}))

const infoOrig = console.info
const errorOrig = console.error
const exitOrig = process.exit

beforeEach(() => {
  console.info = jest.fn()
  console.error = jest.fn()
  process.exit = jest.fn() as any
})

afterEach(() => {
  console.info = infoOrig
  console.error = errorOrig
  process.exit = exitOrig
})

test('ok', () => {
  main()
  expect(execSync.mock.calls).toMatchSnapshot()
})

test('nok', () => {
  execSync.mockImplementation(() => { throw 'Boom!' })
  main()
  expect(console.error).toBeCalledWith('With errors ONE.md')
  expect(process.exit).toBeCalledWith(1)
})
