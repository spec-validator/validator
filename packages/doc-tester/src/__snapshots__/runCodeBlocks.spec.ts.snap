// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ok 1`] = `
Array [
  Array [
    "yarn ts-node -r tsconfig-paths/register",
    Object {
      "input": "

const meaning = 42
",
    },
  ],
  Array [
    "yarn ts-node -r tsconfig-paths/register",
    Object {
      "input": "

const meaning = 42
",
    },
  ],
]
`;
